Source/HTTP/WinHttp/winhttp_http_task.h

```diff
@@ -16,6 +16,39 @@ enum msg_body_type
     transfer_encoding_chunked
 };
 
+class win32_cs
+{
+public:
+    win32_cs() { InitializeCriticalSection(&m_cs); }
+    ~win32_cs() { DeleteCriticalSection(&m_cs); }
+
+    void lock() { EnterCriticalSection(&m_cs); }
+    void unlock() { LeaveCriticalSection(&m_cs); }
+
+private:
+    CRITICAL_SECTION m_cs;
+};
+
+class win32_cs_autolock
+{
+public:
+    win32_cs_autolock(win32_cs* pCS)
+        : m_pCS(pCS)
+    {
+        m_pCS->lock();
+        //HC_TRACE_INFORMATION(HTTPCLIENT, "win32_cs_autolock locked [ID %lu]", GetCurrentThreadId());
```

Comment:
remove? or uncomment if its useful

Reply:
its kinda spammy but useful when watching threads so i think just leaving it like this is fine for now

---
In reply to: [237280338](https://github.com/Microsoft/libHttpClient/pull/339#discussion_r237280338) [](ancestors = 237280338)